_docs-2/development/iterators.md

```diff
@@ -408,12 +408,16 @@ On an instance of an Iterator: `init` is always called before `seek`, `seek` is
 
 ### Teardown
 
-As mentioned, instance of Iterators may be torn down inside of the server transparently. When a complex
-collection of iterators is performing some advanced functionality, they will not be torn down until a Key-Value
-pair is returned out of the "stack" of Iterators (and added into the batch of Key-Values to be returned
-to the caller). Being torn-down is equivalent to a new instance of the Iterator being creating and `deepCopy`
-being called on the new instance with the old instance provided as the argument to `deepCopy`. References
-to the old instance are removed and the object is lazily garbage collected by the JVM.
+As mentioned, instances of iterators may be torn down inside the server transparently. When a 
+complex collection of iterators is performing some advanced functionality, they will not be torn 
+down until a Key-Value pair is returned out of the "stack" of iterators (and added into the batch of
+Key-Values to be returned to the caller). Being torn down means that the iterator stack is recreated
+as new instances with the original options, and then it is seeked from where it left off. References
+to the old instances are removed and the objects are lazily garbage collected by the JVM.
```

Comment:
The sentence about the references to the old instances being garbace collected can be removed.

Reply:
Addressed in fd55b58